src/lib.rs

```diff
@@ -72,3 +80,37 @@ fn enumerate_project_files(language: &dyn SupportedLanguage) -> Vec<DirEntry> {
         .filter(|entry| entry.metadata().unwrap().is_file())
         .collect()
 }
+#[derive(Debug)]
+struct TreeSitterMatcher {
+    matches: Mutex<Vec<Match>>,
+}
+
+impl TreeSitterMatcher {
+    fn new(mut matches: Vec<Match>) -> Self {
+        matches.sort_by_key(|m| m.start());
+
+        Self {
+            matches: Mutex::new(matches),
+        }
+    }
+}
+
+impl Matcher for TreeSitterMatcher {
+    type Captures = NoCaptures;
+
+    type Error = NoError;
+
+    fn find_at(&self, _haystack: &[u8], at: usize) -> Result<Option<Match>, Self::Error> {
+        let _match = self
+            .matches
+            .lock()
+            .unwrap()
+            .pop()
+            .map(|m| Match::new(m.start() - at, m.end() - at));
+        Ok(_match)
+    }
+
+    fn new_captures(&self) -> Result<Self::Captures, Self::Error> {
+        Ok(NoCaptures::new())
```

Comment:
I don't even know what captures do in regular grep.

Reply:
I think captures are when you have `(...)` in your regex pattern and then can reference those specific chunks of the matched text in eg a `.replace()` "replacer" argument

Reply:
Oh yeah, I knew about regex captures in general, but I didn't know grep could do things with them (can it do string replacements?)

Reply:
Ah right I'm conflating `regex`/`ripgrep`, I'm not sure what capture-related "surface area" `ripgrep` exposes?